dist/src/main/java/io/camunda/application/commons/backup/BackupPriorityConfiguration.java

```diff
@@ -102,37 +122,25 @@ private <A> Function<Map<String, A>, String> differentConfigFor(final String fie
 
   @Bean
   public BackupPriorities backupPriorities() {
-    final var indexPrefix =
-        allMatch(
-                NO_CONFIG_ERROR_MESSAGE,
-                differentConfigFor("indexPrefix"),
-                Map.of(
-                    "operate",
-                    Optional.ofNullable(operateProperties).map(OperateProperties::getIndexPrefix),
-                    "tasklist",
-                    Optional.ofNullable(tasklistProperties)
-                        .map(TasklistProperties::getIndexPrefix)),
-                skipEmptyOptional())
-            .get();
-    final var isElasticsearch =
-        allMatch(
-                NO_CONFIG_ERROR_MESSAGE,
-                differentConfigFor("database.type"),
-                Map.of(
-                    "operate",
-                    Optional.ofNullable(operateProperties)
-                        .map(ignored -> DatabaseInfo.isCurrent(DatabaseType.Elasticsearch)),
-                    "tasklist",
-                    Optional.ofNullable(tasklistProperties)
-                        .map(prop -> prop.getDatabase().equals(TasklistProperties.ELASTIC_SEARCH))),
-                skipEmptyOptional())
-            .get();
+    final var indexPrefix = getIndexPrefix();
+
+    final boolean isElasticsearch = getIsElasticsearch();
+
+    final var indexService =
+        new OptimizeIndexNameService(indexPrefix.isBlank() ? "optimize" : indexPrefix);
+
     final List<Prio1Backup> prio1 =
         List.of(
             // OPERATE
             new ImportPositionIndex(indexPrefix, isElasticsearch),
             // TASKLIST
-            new TasklistImportPositionIndex(indexPrefix, isElasticsearch));
+            new TasklistImportPositionIndex(indexPrefix, isElasticsearch),
+            // OPTIMIZE
+            new OptimizePrio1Delegate<>(
+                isElasticsearch
+                    ? new PositionBasedImportIndexES()
+                    : new PositionBasedImportIndexOS(),
+                indexService));
```

Comment:
I think we would also need to include `TimestampBasedImportIndexES` and `TimestampBasedImportIndexOS` in Prio1